internal/ircserver/cmd_mode.go

```diff
@@ -114,6 +114,27 @@ func (i *IRCServer) cmdMode(s *Session, reply *Replyctx, msg *irc.Message) {
 				case 't', 's', 'i', 'n':
 					c.modes[char] = newvalue
 
+				case 'k':
+					if newvalue {
+						if mode.Param == "" {
+							break
+						}
+						c.key = mode.Param
+						i.sendUser(s, reply, &irc.Message{
+							Prefix:  i.ServerPrefix,
+							Command: irc.MODE,
+							Params:  []string{channelname, "+k", mode.Param},
```

Comment:
nit: use c.key instead of mode.Param here for symmetry with the else branch

Reply:
Done.